packages/robusta_firebase_cloud_messaging/lib/src/extension.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+import 'dart:developer';
+
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:meta/meta.dart';
+import 'package:robusta_firebase_core/robusta_firebase_core.dart';
+import 'package:robusta_runner/robusta_runner.dart';
+
+part './permission_settings.dart';
+
+/// {@template robusta_firebase_cloud_messaging}
+/// A Helper class for Firebase Cloud Messaiging
+/// {@endtemplate}
+class FirebaseCloudMessagingExtension implements DependenceExtension {
+  /// {@macro robusta_firebase_cloud_messaging}
+  FirebaseCloudMessagingExtension({
+    required EventManager eventManager,
+    NotiPermissionRequestSettings? settings,
+    NotiRequestStrategy? notiRequestStrategy,
+  })  : _eventManager = eventManager,
+        _notiRequestStrategy = notiRequestStrategy ?? NotiRequestStrategy.init,
+        _settings = settings ?? const NotiPermissionRequestSettings();
+
+  final EventManager _eventManager;
+  final NotiRequestStrategy _notiRequestStrategy;
+  final NotiPermissionRequestSettings _settings;
+
+  @override
+  FutureOr<void> load(Configurator configurator) {
+    configurator.addBoot(_boot, priority: 4095);
+  }
+
+  @override
+  List<Type> dependsOn() => [FirebaseCoreExtension];
+
+  Future<void> _boot(ProviderContainer container) async {
+    if (_notiRequestStrategy == NotiRequestStrategy.init) {
+      await permissionRequest(container);
+    }
+
+    final messaging = FirebaseMessaging.instance;
+
+    final notiAuthStatus = await messaging.getNotificationSettings();
+
+    if (notiAuthStatus.authorizationStatus == AuthorizationStatus.authorized) {
+      FirebaseMessaging.onMessage.listen((RemoteMessage rMessage) async {
```

Comment:
How about in-app messages and background messages?

Reply:
In-app Message should be in a separate extension.